### PR TITLE
[Fluid] Compressible NS slip wall porosity

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**Description**
This PR adds the possibility of considering "porous" slip walls in the compressible NS explicit strategy slip BC imposition. This is quite helpful during the flow initialization.

So far I only add the changes in the strategy as well as the `SLIP_WALL_FACTOR` variable. Eventually, we can create a Python process that sets this value in the `ProcessInfo` in accordance to a user-defined table.

### CODE REVIEW

File: applications/FluidDynamicsApplication/custom_strategies/strategies/compressible_navier_stokes_explicit_solving_strategy_runge_kutta_4.h [hunk 1]
```diff
@@ -507,13 +507,15 @@ class CompressibleNavierStokesExplicitSolvingStrategyRungeKutta4 : public Explic
...
         auto &r_model_part = BaseType::GetModelPart();
+        const auto& r_process_info = r_model_part.GetProcessInfo();
+        const double wall_factor = r_process_info.Has(SLIP_WALL_FACTOR) ? r_process_info[SLIP_WALL_FACTOR] : 1.0;
 
-        block_for_each(r_model_part.Nodes(), [](Node<3>& rNode){
+        block_for_each(r_model_part.Nodes(), [&wall_factor](Node<3>& rNode){
             if (rNode.Is(SLIP)) {
                 auto unit_normal = rNode.FastGetSolutionStepValue(NORMAL);
                 unit_normal /= norm_2(unit_normal);
                 auto& r_mom = rNode.FastGetSolutionStepValue(MOMENTUM);
```
Comment (philbucher) [inline applications/FluidDynamicsApplication/custom_strategies/strategies/compressible_navier_stokes_explicit_solving_strategy_runge_kutta_4.h]:
```suggestion
                array_1d<double, 3>& unit_normal = rNode.FastGetSolutionStepValue(NORMAL);
                unit_normal /= norm_2(unit_normal);
                array_1d<double, 3>& r_mom = rNode.FastGetSolutionStepValue(MOMENTUM);
```

see #7414

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_strategies/strategies/compressible_navier_stokes_explicit_solving_strategy_runge_kutta_4.h]: Thanks for noting it.

File: applications/FluidDynamicsApplication/fluid_dynamics_application_variables.h [hunk 2]
```diff
@@ -94,6 +94,7 @@ KRATOS_DEFINE_APPLICATION_VARIABLE( FLUID_DYNAMICS_APPLICATION, double, ARTIFICI
 KRATOS_DEFINE_APPLICATION_VARIABLE( FLUID_DYNAMICS_APPLICATION, double, ARTIFICIAL_DYNAMIC_VISCOSITY)
 KRATOS_DEFINE_APPLICATION_VARIABLE( FLUID_DYNAMICS_APPLICATION, double, DENSITY_PROJECTION)
 KRATOS_DEFINE_APPLICATION_VARIABLE( FLUID_DYNAMICS_APPLICATION, double, TOTAL_ENERGY_PROJECTION)
+KRATOS_DEFINE_APPLICATION_VARIABLE( FLUID_DYNAMICS_APPLICATION, double, SLIP_WALL_FACTOR)
```
Comment (philbucher) [inline applications/FluidDynamicsApplication/fluid_dynamics_application_variables.h:97]: you are missing `KRATOS_REGISTER_APPLICATION_VARIABLE`

**Comment by rubenzorrilla** [comment]
Closing after #9765.

**Comment by loumalouomega** [comment]
Do we still need this branch @rubenzorrilla ?, can we remove it?

**Pull request closed without merging**